src/term-spec.js

```diff
@@ -128,21 +128,133 @@ declare export class AssignmentTargetPropertyProperty extends AssignmentTargetPr
 // class
 declare export class ClassExpression extends Expression {
   name?: BindingIdentifier;
-  super?: Expression;
-  elements: ClassElement[];
+  typeParameters?: TypeParameterDeclaration[];
+  heritageClauses?: HeritageClause[];
+  elements: (ClassElement | IndexSignatureDeclaration)[];
 }
 
 declare export class ClassDeclaration extends Statement {
+  isAbstract: any; // boolean
   name: BindingIdentifier;
-  super?: Expression;
-  elements: ClassElement[];
+  typeParameters?: TypeParameterDeclaration[];
+  heritageClauses?: HeritageClause[];
+  elements: (ClassElement | IndexSignatureDeclaration)[];
```

Comment:
The `IndexSignatureDeclaration` is handled as a special case since
1. It cannot be static nor have an access modifier;
1. It can be present on interfaces and type literals as well.
There might be a better way of dealing with it.